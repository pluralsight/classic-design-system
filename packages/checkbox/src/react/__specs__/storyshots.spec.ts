import path from 'path'
import initStoryshots from '@storybook/addon-storyshots'

jest.mock('@pluralsight/ps-design-system-util', () => ({
  ...jest.requireActual('@pluralsight/ps-design-system-util'),
  useUniqueId: jest.fn().mockImplementation(prefix => prefix + 'unique-id')
}))

initStoryshots({
  configPath: path.resolve(__dirname, '../../../.storybook'),
  framework: 'react'
})
