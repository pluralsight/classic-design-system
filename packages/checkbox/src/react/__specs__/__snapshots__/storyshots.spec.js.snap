// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Checkbox checked 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "padding": "24px",
      }
    }
  >
    <label
      data-css-kn3wr0=""
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <div
        data-css-vzrwad=""
      >
        <div
          aria-checked={true}
          data-css-1sxo0xr=""
          role="checkbox"
          tabIndex="0"
        >
          <svg
            aria-label="Checkmark"
            data-css-a3uikz=""
            role="img"
            version="1.1"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="6.89667458 13 2.62114014 8.72446556 4.12826603 7.21733967 6.89667458 9.97505938 12.871734 4 14.3788599 5.51781473"
            />
          </svg>
        </div>
      </div>
      <input
        checked={true}
        data-css-gyrv08=""
        disabled={false}
        name="colorRed"
        readOnly={true}
        tabIndex="-1"
        type="checkbox"
        value="red"
      />
      <div
        data-css-pgc406=""
      >
        Red
      </div>
    </label>
  </div>
</div>
`;

exports[`Storyshots Checkbox default 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "padding": "24px",
      }
    }
  >
    <label
      data-css-kn3wr0=""
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <div
        data-css-vzrwad=""
      >
        <div
          aria-checked={false}
          data-css-v2gdam=""
          role="checkbox"
          tabIndex="0"
        />
      </div>
      <input
        checked={false}
        data-css-gyrv08=""
        disabled={false}
        name="colorRed"
        readOnly={true}
        tabIndex="-1"
        type="checkbox"
        value="red"
      />
      <div
        data-css-pgc406=""
      >
        Red
      </div>
    </label>
  </div>
</div>
`;

exports[`Storyshots Checkbox disabled & errored 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "padding": "24px",
      }
    }
  >
    <label
      data-css-1gz58lp=""
      onClick={null}
      onKeyDown={null}
    >
      <div
        data-css-z3vkis=""
      >
        <div
          aria-checked={false}
          data-css-v2gdam=""
          role="checkbox"
          tabIndex="-1"
        />
      </div>
      <input
        checked={false}
        data-css-gyrv08=""
        disabled={true}
        name="colorRed"
        readOnly={true}
        tabIndex="-1"
        type="checkbox"
        value="red"
      />
      <div
        data-css-pgc406=""
      >
        Red
      </div>
    </label>
  </div>
</div>
`;

exports[`Storyshots Checkbox disabled 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "padding": "24px",
      }
    }
  >
    <label
      data-css-1gz58lp=""
      onClick={null}
      onKeyDown={null}
    >
      <div
        data-css-1gypvub=""
      >
        <div
          aria-checked={false}
          data-css-v2gdam=""
          role="checkbox"
          tabIndex="-1"
        />
      </div>
      <input
        checked={false}
        data-css-gyrv08=""
        disabled={true}
        name="colorRed"
        readOnly={true}
        tabIndex="-1"
        type="checkbox"
        value="red"
      />
      <div
        data-css-pgc406=""
      >
        Red
      </div>
    </label>
  </div>
</div>
`;

exports[`Storyshots Checkbox error 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "padding": "24px",
      }
    }
  >
    <div>
      <label
        data-css-kn3wr0=""
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <div
          data-css-19hvvde=""
        >
          <div
            aria-checked={true}
            data-css-1sxo0xr=""
            role="checkbox"
            tabIndex="0"
          >
            <svg
              aria-label="Checkmark"
              data-css-a3uikz=""
              role="img"
              version="1.1"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="6.89667458 13 2.62114014 8.72446556 4.12826603 7.21733967 6.89667458 9.97505938 12.871734 4 14.3788599 5.51781473"
              />
            </svg>
          </div>
        </div>
        <input
          checked={true}
          data-css-gyrv08=""
          disabled={false}
          name="colorRed"
          readOnly={true}
          tabIndex="-1"
          type="checkbox"
          value="red"
        />
        <div
          data-css-pgc406=""
        >
          Red
        </div>
      </label>
      <label
        data-css-kn3wr0=""
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <div
          data-css-19hvvde=""
        >
          <div
            aria-checked={false}
            data-css-v2gdam=""
            role="checkbox"
            tabIndex="0"
          />
        </div>
        <input
          checked={false}
          data-css-gyrv08=""
          disabled={false}
          name="colorRed"
          readOnly={true}
          tabIndex="-1"
          type="checkbox"
          value="red"
        />
        <div
          data-css-pgc406=""
        >
          Red
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`Storyshots Checkbox state demo 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "padding": "24px",
      }
    }
  >
    <div>
      <div
        style={
          Object {
            "color": "#555555",
          }
        }
      >
        Checked:
         
        
      </div>
      <label
        data-css-kn3wr0=""
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <div
          data-css-vzrwad=""
        >
          <div
            aria-checked={false}
            data-css-v2gdam=""
            role="checkbox"
            tabIndex="0"
          />
        </div>
        <input
          checked={false}
          data-css-gyrv08=""
          disabled={false}
          name="colorRed"
          readOnly={true}
          tabIndex="-1"
          type="checkbox"
          value="red"
        />
        <div
          data-css-pgc406=""
        >
          Red
        </div>
      </label>
      <label
        data-css-kn3wr0=""
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <div
          data-css-vzrwad=""
        >
          <div
            aria-checked={false}
            data-css-v2gdam=""
            role="checkbox"
            tabIndex="0"
          />
        </div>
        <input
          checked={false}
          data-css-gyrv08=""
          disabled={false}
          name="colorGreen"
          readOnly={true}
          tabIndex="-1"
          type="checkbox"
          value="green"
        />
        <div
          data-css-pgc406=""
        >
          Green
        </div>
      </label>
      <label
        data-css-kn3wr0=""
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <div
          data-css-vzrwad=""
        >
          <div
            aria-checked={false}
            data-css-v2gdam=""
            role="checkbox"
            tabIndex="0"
          />
        </div>
        <input
          checked={false}
          data-css-gyrv08=""
          disabled={false}
          name="colorBlue"
          readOnly={true}
          tabIndex="-1"
          type="checkbox"
          value="blue"
        />
        <div
          data-css-pgc406=""
        >
          Blue
        </div>
      </label>
    </div>
  </div>
</div>
`;
